src/sentry/testutils/factories.py

```diff
@@ -385,6 +387,16 @@ def create_api_key(organization, scope_list=None, **kwargs):
             organization_id=organization.id if organization else None, scope_list=scope_list
         )
 
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.CONTROL)
+    def create_auth_provider(**kwargs):
+        return AuthProvider.objects.create(**kwargs)
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.CONTROL)
+    def create_auth_identity(**kwargs):
+        return AuthIdentity.objects.create(**kwargs)
+
```

Comment:
Going to make these whenever I get the chance since they often remove needing `with assume_test_silo_mode` from the actual test.